src/aind_metadata_upgrader/procedures_upgrade.py

```diff
@@ -0,0 +1,449 @@
+            "headframe_material": get_or_default(old_subj_procedure, Headframe, "headframe_material"),
+            "well_part_number": get_or_default(old_subj_procedure, Headframe, "well_part_number"),
+            "well_type": get_or_default(old_subj_procedure, Headframe, "well_type"),
+        }
+
+        return construct_new_model(headframe_dict, Headframe, self.allow_validation_errors)
+
+    def upgrade_nanoject_injection(self, old_subj_procedure: dict):
+        """Map legacy NanojectInjection model to current version"""
+
+        injection_dict = {
+            "injection_volume": old_subj_procedure.get("injection_volume", None),
+            "injection_volume_unit": get_or_default(old_subj_procedure, NanojectInjection, "injection_volume_unit"),
+            "injection_materials": old_subj_procedure.get("injection_materials", [None]),
+            "recovery_time": old_subj_procedure.get("recovery_time", None),
+            "recovery_time_unit": get_or_default(old_subj_procedure, NanojectInjection, "recovery_time_unit"),
+            "injection_duration": old_subj_procedure.get("injection_duration", None),
+            "injection_duration_unit": get_or_default(old_subj_procedure, NanojectInjection, "injection_duration_unit"),
+            "instrument_id": old_subj_procedure.get("instrument_id", None),
+            "protocol_id": old_subj_procedure.get("protocol_id", "dx.doi.org/10.17504/protocols.io.bgpujvnw"),
+            "injection_coordinate_ml": old_subj_procedure.get("injection_coordinate_ml", None),
+            "injection_coordinate_ap": old_subj_procedure.get("injection_coordinate_ap", None),
+            "injection_coordinate_depth": old_subj_procedure.get("injection_coordinate_depth", None),
+            "injection_coordinate_unit": get_or_default(
+                old_subj_procedure, NanojectInjection, "injection_coordinate_unit"
+            ),
+            "injection_coordinate_reference": old_subj_procedure.get("injection_coordinate_reference", None),
+            "bregma_to_lambda_distance": old_subj_procedure.get("bregma_to_lambda_distance", None),
+            "bregma_to_lambda_unit": get_or_default(old_subj_procedure, NanojectInjection, "bregma_to_lambda_unit"),
+            "injection_angle": old_subj_procedure.get("injection_angle", None),
+            "injection_angle_unit": get_or_default(old_subj_procedure, NanojectInjection, "injection_angle_unit"),
+            "targeted_structure": old_subj_procedure.get("targeted_structure", None),
+            "injection_hemisphere": old_subj_procedure.get("injection_hemisphere", None),
+        }
+
+        return construct_new_model(injection_dict, NanojectInjection, self.allow_validation_errors)
+
+    def upgrade_perfusion(self, old_subj_procedure: dict):
+        """Map legacy Perfusion model to current version"""
+
+        perfusion_dict = {
+            "protocol_id": old_subj_procedure.get("protocol_id", "dx.doi.org/10.17504/protocols.io.bg5vjy66"),
```

Comment:
Can you make this a class constant?

`DEFAULT_PERFUSION_PROTOCOL = "dx.doi.org/10.17504/protocols.io.bg5vjy66"`

And then use that here.

Reply:
Do you want this within the class or within the script?